rosbag2_cpp/src/rosbag2_cpp/writers/sequential_writer.cpp

```diff
@@ -77,6 +78,7 @@ void SequentialWriter::init_metadata()
   file_info.message_count = 0;
   metadata_.custom_data = storage_options_.custom_data;
   metadata_.files = {file_info};
+  metadata_.ros_distro = rcpputils::get_env_var("ROS_DISTRO");
```

Comment:
Could we print a warning (or possibly exit) if ROS_DISTRO is not set? Users might try to launch a recorder outside of `ros2 launch` or a ros env, and may not be aware of the dependency here.

Reply:
I definitely don't want it to exit (especially if everything else would work properly!). But since unset ROS_DISTRO will be a minority case, I see no problem with a warning message